### ISSUE TITLE
zammad-postgresql_1     | FATAL:  role "zammad" does not exist

### ISSUE THREAD

**Issue by strickes** [zammad/zammad-docker-compose#132]
### Infos:

* Docker version: Docker version 19.03.5, build 633a0ea838
* Docker-compose version: docker-compose version 1.25.3, build d4d1b42b
* Operating system (Docker host): Ubuntu 18.04 (Linux web06 4.15.0-62-generic #69-Ubuntu SMP Wed Sep 4 20:55:53 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux)

### Expected behavior:

no startup when using the latest docker-compose. I upgraded from the previous versions via docker-compose pull

### Actual behavior:
I changed the environment variable to match the latest postgres:
environment:
- POSTGRES_HOST_AUTH_METHOD=trust
according to https://github.com/zammad/zammad-docker-compose/issues/128

but after upgrading I still get the following error in the log:

zammad-postgresql_1     | FATAL:  role "zammad" does not exist
zammad-init_1           | FATAL:  role "zammad" does not exist
zammad-init_1           | Couldn't create 'zammad_production' database. Please check your configuration.
zammad-init_1           | rake aborted!
zammad-init_1           | ActiveRecord::NoDatabaseError: FATAL:  role "zammad" does not exist

### Steps to reproduce the behavior:

*

**Comment by abeluck** [comment]
> I changed the environment variable to match the latest postgres:
> environment:
> * POSTGRES_HOST_AUTH_METHOD=trust
    according to #128

That is not a correct fix. @theweirdone who told you to do that isn't a maintainer on this project. And it is quite an insecure way of fixing the issue.

@monotek, the maintainer of this project, chose to fix it in a better way see [this commit](https://github.com/zammad/zammad-docker-compose/commit/607fa4a02a916d2d3f247297aa90f40f3f7093d5)

**Comment by Waasil** [comment]
I have the same problem. I have followed steps from [DOCS](https://docs.zammad.org/en/latest/install/docker-compose.html). Should I customize something on my own before command `docker-compose up` ?

**Comment by strickes** [comment]
I edited the files locally but no change. I still got

zammad-postgresql_1     | PostgreSQL Database directory appears to contain a database; Skipping initialization
zammad-postgresql_1     | 
zammad-postgresql_1     | LOG:  database system was shut down at 2020-02-25 13:14:10 UTC
zammad-postgresql_1     | LOG:  MultiXact member wraparound protections are now enabled
zammad-postgresql_1     | LOG:  database system is ready to accept connections
zammad-postgresql_1     | LOG:  autovacuum launcher started
zammad-postgresql_1     | LOG:  incomplete startup packet
zammad-postgresql_1     | FATAL:  role "zammad" does not exist

**Comment by Waasil** [comment]
I hope that somebody will answere here because my issue #133 was tagged as duplicate and there is still no solution for that problem neither here nor there.

**Comment by monotek** [comment]
The zammad user has to be created as the init script of the postgres docker container will not run when a database already exists.

Try:

```
docker-compose up
docker exec -it zammaddockercompose_zammad-postgresql_1 bash
psql --username postgres --dbname zammad_production
CREATE USER zammad;
ALTER USER zammad WITH PASSWORD 'zammad';
ALTER USER zammad WITH SUPERUSER CREATEDB;
```

**Comment by AdamWasilewski** [comment]
> The zammad user has to be created as the init script of the postgres docker container will not run when a database already exists.
> 
> Try:
> 
> ```
> docker-compose up
> docker exec -it zammaddockercompose_zammad-postgresql_1 bash
> psql --username postgres --dbname zammad_production
> CREATE USER zammad;
> ALTER USER zammad WITH PASSWORD 'zammad';
> ALTER USER zammad WITH SUPERUSER CREATEDB;
> ```

Does it mean that I need to prepare my own postgresql database on server where I launch docker-compose?

**Comment by monotek** [comment]
no

**Comment by Waasil** [comment]
Thank you monotek for idea :)

I fixed `role "zammad" does not exist` in that way:

- `docker-compose up`
- After some logs I stopped all containers by CTRL + C
- Started container with Postgresql and created `zammad` user in psql (and granted all privileges)
- Stopped container with Postgresql
- `docker-compose up` 

Everything works.

**Comment by GuidoDr** [comment]
I had the same problem after I just did an update from the previous zammd-docker-compose version to this new version.
I now did 

> docker-compose up
> docker exec -it zammaddockercompose_zammad-postgresql_1 bash
> psql --username postgres --dbname zammad_production
> CREATE USER zammad;
> ALTER USER zammad WITH PASSWORD 'zammad';
> ALTER USER zammad WITH SUPERUSER CREATEDB;

But still Zammad is not running, I do now see an error in zammad-elasticsearch which is now always restarting. How can I fix this?

zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-26T18:13:53,362Z", "level": "INFO", "component": "o.e.t.TransportService", "cluster.name": "docker-cluster", "node.name": "0e203f14aa0f", "message": "publish_address {172.18.0.46:9300}, bound_addresses {0.0.0.0:9300}" }
zammad-elasticsearch_1  | {"type": "server", "timestamp": "2020-02-26T18:13:53,409Z", "level": "**ERROR**", "component": "o.e.b.ElasticsearchUncaughtExceptionHandler", "cluster.name": "docker-cluster", "node.name": "0e203f14aa0f", "message": "uncaught exception in thread [main]",
zammad-elasticsearch_1  | "stacktrace": ["org.elasticsearch.bootstrap.StartupException: java.lang.IllegalStateException: **The index [[zammad_production/42WtFPMyR0GmpQ9IMO2zGw]] was created with version [5.6.5] but the minimum compatible version is [6.0.0-beta1]. It should be re-indexed in Elasticsearch 6.x before upgrading to 7.6.0."**,
zammad-elasticsearch_1  | "at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:174) ~[elasticsearch-7.6.0.jar:7.6.0]",
zammad-elasticsearch_1  | "at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:161) ~[elasticsearch-7.6.0.jar:7.6.0]",
zammad-elasticsearch_1  | "at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86) ~[elasticsearch-7.6.0.jar:7.6.0]",
zammad-elasticsearch_1  | "at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:125) ~[elasticsearch-cli-7.6.0.jar:7.6.0]",
zammad-elasticsearch_1  | "at org.elasticsearch.cli.Command.main(Command.java:90) ~[elasticsearch-cli-7.6.0.jar:7.6.0]",
zammad-elasticsearch_1  | "at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:126) ~[elasticsearch-7.6.0.jar:7.6.0]",
zammad-elasticsearch_1  | "at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92) ~[elasticsearch-7.6.0.jar:7.6.0]",
zammad-elasticsearch_1  | "Caused by: java.lang.IllegalStateException: The index [[zammad_production/42WtFPMyR0GmpQ9IMO2zGw]] was created with version [5.6.5] but the minimum compatible version is [6.0.0-beta1]. It should be re-indexed in Elasticsearch 6.x before upgrading to 7.6.0.",

**Comment by MrGeneration** [comment]
@GuidoDr 

```
The index [[zammad_production/42WtFPMyR0GmpQ9IMO2zGw]] was created with version [5.6.5] but the minimum compatible version is [6.0.0-beta1]. It should be re-indexed in Elasticsearch 6.x before upgrading to 7.6.0.
```

Wrong issue. Please be a bit more carefull.
Also, have a look at this readme that will help you to fix this issue:

https://github.com/zammad/zammad-docker-compose/blob/master/README.md#from--320-12

**Comment by monotek** [comment]
That's another problem. See: https://github.com/zammad/zammad-docker-compose/blob/master/README.md#from--320-12

**Comment by GuidoDr** [comment]
> 
> 
> @GuidoDr That's another problem. See: https://github.com/zammad/zammad-docker-compose/blob/master/README.md#from--320-12

I know, I already solved it. MrGeneration had already mentioned the same link like you two hours earlier.
I had posted it here since it first started with the missing zammad role and after I had solved due to the information above, I thenencountered the next problem. I had not read the README since I normally only I only do a docker-compose pull for every new version and never had any problems before in the last 1-2 years. Next time when I do encounter a problem I will first have a look at the README file. ;-)
Thanks both of your for your help.
